health/health_test.go

```diff
@@ -0,0 +1,352 @@
+package health
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+
+	fs "github.com/featureform/filestore"
+	"github.com/featureform/metadata"
+	pc "github.com/featureform/provider/provider_config"
+	pt "github.com/featureform/provider/provider_type"
+	"github.com/joho/godotenv"
+	"go.uber.org/zap/zaptest"
+)
+
+var providerType = flag.String("provider", "", "provider type under test")
+
+type testMember struct {
+	providerDef metadata.ProviderDef
+}
+
+func checkEnv(envVar string) string {
+	value, has := os.LookupEnv(envVar)
+	if !has {
+		panic(fmt.Sprintf("Environment variable not found: %s", envVar))
+	}
+	return value
+}
+
+func initSpark(t *testing.T, executorType pc.SparkExecutorType, storeType fs.FileStoreType) (pc.SerializedConfig, pc.SparkConfig) {
+	var executorConfig pc.SparkExecutorConfig
+
+	switch executorType {
+	case pc.SparkGeneric:
+		executorConfig = &pc.SparkGenericConfig{
+			Master:        os.Getenv("GENERIC_SPARK_MASTER"),
+			DeployMode:    os.Getenv("GENERIC_SPARK_DEPLOY_MODE"),
+			PythonVersion: os.Getenv("GENERIC_SPARK_PYTHON_VERSION"),
+		}
+	case pc.Databricks:
+		executorConfig = &pc.DatabricksConfig{
+			Host:    os.Getenv("DATABRICKS_HOST"),
+			Token:   os.Getenv("DATABRICKS_TOKEN"),
+			Cluster: os.Getenv("DATABRICKS_CLUSTER"),
+		}
+	case pc.EMR:
+		executorConfig = &pc.EMRConfig{
+			Credentials: pc.AWSCredentials{
+				AWSAccessKeyId: os.Getenv("AWS_ACCESS_KEY_ID"),
+				AWSSecretKey:   os.Getenv("AWS_SECRET_KEY"),
+			},
+			ClusterRegion: os.Getenv("AWS_EMR_CLUSTER_REGION"),
+			ClusterName:   os.Getenv("AWS_EMR_CLUSTER_ID"),
+		}
+	default:
+		t.Fatalf("Invalid executor type: %v", executorType)
+	}
+
+	var fileStoreConfig pc.SparkFileStoreConfig
+	switch storeType {
+	case fs.S3:
+		fileStoreConfig = &pc.S3FileStoreConfig{
+			Credentials: pc.AWSCredentials{
+				AWSAccessKeyId: os.Getenv("AWS_ACCESS_KEY_ID"),
+				AWSSecretKey:   os.Getenv("AWS_SECRET_KEY"),
+			},
+			BucketRegion: os.Getenv("S3_BUCKET_REGION"),
+			BucketPath:   os.Getenv("S3_BUCKET_PATH"),
+			Path:         os.Getenv(""),
+		}
+	case fs.GCS:
+		credsFile := os.Getenv("GCP_CREDENTIALS_FILE")
+		content, err := ioutil.ReadFile(credsFile)
+		if err != nil {
+			t.Errorf("Error when opening file: %v", err)
+		}
+		var creds map[string]interface{}
+		err = json.Unmarshal(content, &creds)
+		if err != nil {
+			t.Errorf("Error during Unmarshal() creds: %v", err)
+		}
+
+		fileStoreConfig = &pc.GCSFileStoreConfig{
+			BucketName: os.Getenv("GCS_BUCKET_NAME"),
+			BucketPath: "",
+			Credentials: pc.GCPCredentials{
+				ProjectId: os.Getenv("GCP_PROJECT_ID"),
+				JSON:      creds,
+			},
+		}
+	case fs.Azure:
```

Comment:
should we add HDFS?

Reply:
@sdreyer? I can add it if you think we need to support it in this first phase. We can also add support later as needed.